hbase-common/src/main/java/org/apache/hadoop/hbase/TagBuilder.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright The Apache Software Foundation
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase;
+
+import java.nio.ByteBuffer;
+import org.apache.yetus.audience.InterfaceAudience;
+
+/**
+ * Builder method to create {@link Tag}.
+ */
+@InterfaceAudience.LimitedPrivate(HBaseInterfaceAudience.COPROC)
+public interface TagBuilder {
+  /**
+   * Set type of the tag.
+   * @param type type of the tag
+   * @return {@link TagBuilder}
+   */
+  TagBuilder setType(byte type);
+
+  /**
+   * For {@link ArrayBackedTag}, set the value of the tag.
+   * @param tagBytes tag bytes.
+   * @return {@link TagBuilder}
+   */
+  TagBuilder setTagValue(byte[] tagBytes);
```

Comment:
UnsupportedOperationException is unchecked exception since it extends RunTimeException.  I am little bit hesitant to add it to throws clause.
Referring to this document https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html and the last line:
`Here's the bottom line guideline: If a client can reasonably be expected to recover from an exception, make it a checked exception. If a client cannot do anything to recover from the exception, make it an unchecked exception.`
I don't think in this case client can recover if setTagValue throws UnsupportedOperationException.

@virajjasani  But if you feel strong about this point then I am open to make the changes. Thank you for the comment !

Reply:
Oh I missed the fact that this is already RTE, my bad. If it was not RTE, it would have already been added. Darn it 🤦

I think we are good here, maybe we can just add the expectation in javadoc, should be good enough. Thanks @shahrs87

Reply:
Changed javadoc for both methods setTagValue and setTagByteBuffer. Please review again. Thank you @virajjasani  !